server/src/main/java/org/elasticsearch/script/BucketAggregationScript.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.script;
+
+import java.util.Map;
+
+/**
+ * A script used in bucket aggregations.
+ */
+public abstract class BucketAggregationScript {
+
+    public static final String[] PARAMETERS = { "params" };
+
+    public static final ScriptContext<Factory> CONTEXT = new ScriptContext<>("aggs_executable", Factory.class);
+
+    public abstract Object execute(Map<String, Object> params);
```

Comment:
The intention of adding script contexts was to avoid having generic object returns like this that must be cast. It is especially relevant in numeric processing, as boxing/unboxing is not free. I think we actually want aggregation script classes for each different aggregation use case.

Reply:
@rjernst makes sense :) I was wondering whether or not that (compiling for a specific return type) would be possible with Painless (especially given this kind of thing https://github.com/elastic/elasticsearch/pull/32134).
Thanks for answering :) => on it splitting this up some more.